web/app/js/components/ResourceDetail.jsx

```diff
@@ -181,8 +183,20 @@ export class ResourceDetailBase extends React.Component {
           resourceIsMeshed = _get(this.state.resourceMetrics, '[0].pods.meshedPods') > 0;
         }
 
+        let isTcpOnly = false;
+        let metric = resourceMetrics[0];
+        if (!_isEmpty(metric) && !_isEmpty(metric.tcp)) {
+          let { tcp } = metric;
+
+          if (_isNil(metric.stats) &&
+            (tcp.openConnections > 0 || tcp.readBytes > 0 || tcp.writeBytes > 0)) {
+            isTcpOnly = true;
+          }
+        }
+
+        // figure out when the last traffic this resource received was so we can show a no traffic message
         let lastMetricReceivedTime = this.state.lastMetricReceivedTime;
-        if (!_isEmpty(resourceMetrics[0]) && resourceMetrics[0].totalRequests !== 0) {
+        if ((!_isEmpty(metric) && metric.totalRequests !== 0) || isTcpOnly) {
```

Comment:
i was struggline a little to parse this, what do you think about...
```js
let hasHttp = false;
let hasTcp = false;

let metric = resourceMetrics[0];
if (!_isEmpty(metric)) {
  hasHttp = !_isNil(metric.stats) && metric.totalRequests !== 0;

  if (!_isEmpty(metric.tcp)) {
    let { tcp } = metric;
    hasTcp = tcp.openConnections > 0 || tcp.readBytes > 0 || tcp.writeBytes > 0;
  }
}

// figure out when the last traffic this resource received was so we can show a no traffic message
let lastMetricReceivedTime = this.state.lastMetricReceivedTime;
if (hasHttp || hasTcp) {
  lastMetricReceivedTime = Date.now();
}

let isTcpOnly = !hasHttp && hasTcp;
```

Reply:
wow yes, those `)` were misplaced; good spot!

Reply:
ooh, I like this suggestion, much clearer to read!